rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -70,32 +71,31 @@
 
     @RequestMapping(value = PROJECTS_URL, method = RequestMethod.GET)
     public ResponseEntity<Resources<Resource<Project>>> getProjectsForUser(@RequestParam(value = "name", required = false) String name,
-    		OAuth2Authentication oAuth2Authentication) {
+                                                                           @RequestParam(value = "type", required = false) String projectType,
+                                                                           OAuth2Authentication oAuth2Authentication) {
+
         User sw360User = restControllerHelper.getSw360UserFromAuthentication(oAuth2Authentication);
         List<Project> sw360Projects = new ArrayList<>();
-        if ((name != null) && (!name.isEmpty()))	{
+        if (name != null && !name.isEmpty()) {
             sw360Projects.addAll(projectService.searchProjectByName(name, sw360User));
         } else {
-        	sw360Projects.addAll(projectService.getProjectsForUser(sw360User));
+            sw360Projects.addAll(projectService.getProjectsForUser(sw360User));
         }
 
         List<Resource<Project>> projectResources = new ArrayList<>();
-        for (Project sw360Project : sw360Projects) {
-            // TODO Kai Tödter 2017-01-04
-            // Find better way to decrease details in list resources,
-            // e.g. apply projections or Jackson Mixins
-            sw360Project.setDescription(null);
-            sw360Project.setType(null);
-            sw360Project.setCreatedOn(null);
-            sw360Project.setReleaseIdToUsage(null);
-            sw360Project.setExternalIds(null);
-            sw360Project.setBusinessUnit(null);
+        sw360Projects.stream()
+                .filter(projectType == null ? p -> true : project -> projectType.equals(project.projectType.name()))
```

Comment:
`project -> projectType == null || projectType.equals(project.projectType.name())`

Reply:
updated filter condition for projects and components